Funcky.Test/Extensions/EnumerableExtensions/ZipLongestTest.cs

```diff
@@ -0,0 +1,70 @@
+using System.Linq;
+using Funcky.Extensions;
+using Funcky.Test.TestUtils;
+using Xunit;
+using static Funcky.Functional;
+
+namespace Funcky.Test.Extensions.EnumerableExtensions
+{
+    public sealed class ZipLongestTest
+    {
+        [Fact]
+        public void ZipLongestIsEnumeratedLazily()
+        {
+            var doNotEnumerate = new FailOnEnumerateSequence<object>();
+
+            _ = doNotEnumerate.ZipLongest(doNotEnumerate, (_) => Unit.Value);
+        }
+
+        [Fact]
+        public void GivenTwoEmptySequencesZipLongestReturnsAnEmptySequence()
+        {
+            var numbers = Enumerable.Empty<int>();
+            var strings = Enumerable.Empty<string>();
+
+            var zipped = numbers.ZipLongest(strings, (_) => Unit.Value);
+
+            Assert.Empty(zipped);
+        }
+
+        [Fact]
+        public void GivenTwoSequencesOfTheSameLengthWeGetNoNoneValue()
+        {
+            var numbers = Enumerable.Range(0, 3);
+            var strings = new[] { "Alpha", "Beta", "Gamma" };
+
+            var zipped = numbers
+                .ZipLongest(strings)
+                .ToList();
+
+            Assert.Equal(3, zipped.Count);
+            foreach (var value in zipped)
+            {
+                Assert.True(value.Match(False, False, (_, _) => true));
```

Comment:
Maybe we need a binary `False` and `True` function

Reply:
Added binary constant bool functions
1cdd0fdf1d29192d7d0a26d9df3818070cb7cf8a